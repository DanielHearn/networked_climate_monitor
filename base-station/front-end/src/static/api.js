import { HTTP } from './http-common'

/**
 * Get account details from API
 * @param {string} accessToken - User's access token
 */
export const getAccount = accessToken => {
  const promise = new Promise(function(resolve, reject) {
    HTTP.get('account', {
      headers: { Authorization: 'Bearer ' + accessToken }
    })
      .then(response => {
        resolve(response)
      })
      .catch(e => {
        reject(e)
      })
  })
  return promise
}

/**
 * Update account details in API
 * @param {string} accessToken - User access token
 * @param {object} data - Updated user data
 */
export const patchAccount = (accessToken, data) => {
  const promise = new Promise(function(resolve, reject) {
    HTTP.patch('account', data, {
      headers: { Authorization: 'Bearer ' + accessToken }
    })
      .then(response => {
        resolve(response)
      })
      .catch(e => {
        reject(e)
      })
  })
  return promise
}

/**
 * Attempt user login with API
 * @param {string} email - User's email
 * @param {string} password - User's password
 */
export const login = (email, password) => {
  const promise = new Promise(function(resolve, reject) {
    HTTP.post('login', {
      email: email,
      password: password
    })
      .then(response => {
        resolve(response)
      })
      .catch(e => {
        reject(e)
      })
  })
  return promise
}

/**
 * Attempt user registration with API
 * @param {string} email - User's email
 * @param {string} password - User's password
 */
export const register = (email, password) => {
  const promise = new Promise(function(resolve, reject) {
    HTTP.post('account', {
      email: email,
      password: password
    })
      .then(response => {
        resolve(response)
      })
      .catch(e => {
        reject(e)
      })
  })
  return promise
}

/**
 * Change the user's password
 * @param {string} resetToken - User's reset token
 * @param {string} password - User's password
 */
export const changePassword = (resetToken, password) => {
  const promise = new Promise(function(resolve, reject) {
    HTTP.post('accounts/actions/change-password', {
      reset_token: resetToken,
      password: password
    })
      .then(response => {
        resolve(response)
      })
      .catch(e => {
        reject(e)
      })
  })
  return promise
}

/**
 * Blacklist the access token from future logins
 * @param {string} accessToken - User's access token
 */
export const logoutAccessToken = accessToken => {
  const promise = new Promise(function(resolve, reject) {
    HTTP.post(
      'logout/access',
      {},
      {
        headers: { Authorization: 'Bearer ' + accessToken }
      }
    )
      .then(response => {
        resolve(response)
      })
      .catch(e => {
        reject(e)
      })
  })
  return promise
}

/**
 * Blacklist the refresh token from future logins
 * @param {string} accessToken - User's access token
 */
export const logoutRefreshToken = accessToken => {
  const promise = new Promise(function(resolve, reject) {
    HTTP.post(
      'logout/refresh',
      {},
      {
        headers: { Authorization: 'Bearer ' + accessToken }
      }
    )
      .then(response => {
        resolve(response)
      })
      .catch(e => {
        reject(e)
      })
  })
  return promise
}

/**
 * Update sensor details in API
 * @param {string} accessToken - User's acess token
 * @param {number} sensorID - Sensor ID
 * @param {object} data - Updated sensor data
 */
export const patchSensor = (accessToken, sensorID, data) => {
  const promise = new Promise(function(resolve, reject) {
    HTTP.patch(`sensors/${sensorID}`, data, {
      headers: { Authorization: 'Bearer ' + accessToken }
    })
      .then(response => {
        resolve(response)
      })
      .catch(e => {
        reject(e)
      })
  })
  return promise
}

/**
 * Get climate data for a specified sensor ID from the API
 * @param {string} accessToken - User's access token
 * @param {number} sensorID - Sensor ID
 * @param {number} quantity - Quantity of climate data points to be retrieved
 * @param {date} rangeStart - Start date of the time period
 * @param {date} rangeEnd - End date of the time period
 */
export const getClimateData = (
  accessToken,
  sensorID,
  quantity,
  rangeStart,
  rangeEnd
) => {
  const promise = new Promise(function(resolve, reject) {
    HTTP.get(
      `sensors/${sensorID}/climate-data?range_start=${rangeStart}&range_end=${rangeEnd}`,
      {
        headers: { Authorization: 'Bearer ' + accessToken }
      }
    )
      .then(response => {
        resolve(response)
      })
      .catch(e => {
        reject(e)
      })
  })
  return promise
}

/**
 * Delete the specified sensor using its sensor ID
 * @param {string} accessToken - User's access token
 * @param {number} sensorID - Sensor ID
 */
export const deleteSensor = (accessToken, sensorID) => {
  const promise = new Promise(function(resolve, reject) {
    HTTP.delete(`sensors/${sensorID}`, {
      headers: { Authorization: 'Bearer ' + accessToken }
    })
      .then(response => {
        resolve(response)
      })
      .catch(e => {
        reject(e)
      })
  })
  return promise
}

/**
 * Delete the climate data from the specified sensor using its sensor ID
 * @param {string} accessToken - User's access token
 * @param {number} sensorID - Sensor ID
 */
export const deleteClimateData = (accessToken, sensorID) => {
  const promise = new Promise(function(resolve, reject) {
    HTTP.delete(`sensors/${sensorID}/climate-data`, {
      headers: { Authorization: 'Bearer ' + accessToken }
    })
      .then(response => {
        resolve(response)
      })
      .catch(e => {
        reject(e)
      })
  })
  return promise
}

/**
 * Get all sensors from the API
 * @param {string} accessToken - User's access token
 */
export const getSensors = accessToken => {
  const promise = new Promise(function(resolve, reject) {
    HTTP.get('sensors', {
      headers: { Authorization: 'Bearer ' + accessToken }
    })
      .then(response => {
        resolve(response)
      })
      .catch(e => {
        reject(e)
      })
  })
  return promise
}
