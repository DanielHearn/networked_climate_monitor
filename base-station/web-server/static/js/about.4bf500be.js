(window["webpackJsonp"]=window["webpackJsonp"]||[]).push([["about"],{be4d:function(t,e,n){"use strict";n.r(e);var u=function(){var t=this,e=t.$createElement;t._self._c;return t._m(0)},r=[function(){var t=this,e=t.$createElement,n=t._self._c||e;return n("div",[n("p",[t._v("404 - Not Found")])])}],a=n("2877"),l={},s=Object(a["a"])(l,u,r,!1,null,null,null);e["default"]=s.exports},f820:function(t,e,n){"use strict";n.r(e);var u=function(){var t=this,e=t.$createElement;t._self._c;return t._m(0)},r=[function(){var t=this,e=t.$createElement,n=t._self._c||e;return n("div",{staticClass:"about"},[n("h1",[t._v("This is an about page")])])}],a=n("2877"),l={},s=Object(a["a"])(l,u,r,!1,null,null,null);e["default"]=s.exports}}]);
//# sourceMappingURL=about.4bf500be.js.map