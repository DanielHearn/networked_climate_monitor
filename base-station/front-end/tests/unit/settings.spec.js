import Vuex from 'vuex'
import { mount, createLocalVue } from '@vue/test-utils'
import Settings from './../../src/views/Settings.vue'
import { storeConfig } from './../../src/store/index.js'
import { patchAccount, getAccount } from './../../src/static/api.js'
import { cloneDeep } from 'lodash'

const localVue = createLocalVue()
localVue.use(Vuex)
jest.mock('./../../src/static/api.js')

let wrapper
let store

const accountData = {
  status: 'Account successfully retrieved',
  account: {
    settings:
      "{'temperature_unit':'f','measurement_interval':'30_min','wifi':{'ssid':'test-wifi','password':'test-password'}}",
    id: 1,
    reset_token: '1234',
    email: 'email@email.com'
  }
}

const patchAccountData = {
  status: 'Account successfully updated'
}

const factory = () => {
  store = new Vuex.Store(storeConfig)
  store.state.user.logged_in = true
  store.state.user.access_token = '1234'
  store.state.user.refresh_token = '1234'
  store.state.user.reset_token = '1234'
  store.state.user.email = 'email@email.com'
  store.state.user.settings = {
    temperature_unit: 'c',
    measurement_interval: '10_min',
    wifi: {
      ssid: 'test-wifi',
      password: 'test-password'
    }
  }

  return mount(Settings, {
    mocks: {
      $toasted: {
        show: jest.fn()
      },
      $router: {
        push: jest.fn()
      },
      patchAccount,
      getAccount
    },
    store,
    localVue
  })
}

describe('Settings.vue', () => {
  beforeEach(() => {
    wrapper = null
  })

  afterEach(() => {
    wrapper.destroy()
    jest.resetAllMocks()
  })

  it('has the expected html structure', async () => {
    const accountResponse = cloneDeep(accountData)
    getAccount.mockResolvedValueOnce(accountResponse)

    wrapper = factory()

    await wrapper.vm.$nextTick()

    expect(wrapper.element).toMatchSnapshot()
  })

  it('category changing', async () => {
    const accountResponse = cloneDeep(accountData)
    getAccount.mockResolvedValueOnce(accountResponse)

    wrapper = factory()

    await wrapper.vm.$nextTick()

    let activeCategoryListItem = wrapper.find('.side-panel .list-item.active')
    expect(activeCategoryListItem.find('.heading').text()).toBe(
      'Measurement Settings'
    )
    expect(wrapper.vm.activeCategoryID).toBe(0)
    wrapper
      .findAll('.side-panel .list-item')
      .at(1)
      .find('.actions .button')
      .trigger('click')

    await wrapper.vm.$nextTick()

    activeCategoryListItem = wrapper.find('.side-panel .list-item.active')
    expect(activeCategoryListItem.find('.heading').text()).toBe(
      'Account Management'
    )
    expect(wrapper.vm.activeCategoryID).toBe(1)

    expect(wrapper.element).toMatchSnapshot()
  })

  it('change temperature unit', async () => {
    const accountResponse = cloneDeep(accountData)
    getAccount.mockResolvedValueOnce(accountResponse)
    const patchResponse = cloneDeep(patchAccountData)
    patchAccount.mockResolvedValueOnce(patchResponse)

    wrapper = factory()

    await wrapper.vm.$nextTick()

    expect(wrapper.vm.settings.temperature_unit).toBe('c')

    wrapper.find('label[for="temp_unit_f"]').trigger('click')

    const stringifiedSettings = JSON.stringify(wrapper.vm.settings)
    expect(wrapper.vm.settings.temperature_unit).toBe('f')
    expect(patchAccount).toBeCalledWith(store.state.user.access_token, {
      settings: stringifiedSettings
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('change measurement interval', async () => {
    const accountResponse = cloneDeep(accountData)
    getAccount.mockResolvedValueOnce(accountResponse)
    const patchResponse = cloneDeep(patchAccountData)
    patchAccount.mockResolvedValueOnce(patchResponse)

    wrapper = factory()

    await wrapper.vm.$nextTick()

    expect(wrapper.vm.settings.measurement_interval).toBe('10_min')

    expect(
      wrapper.find('.measurement-interval-setting .text').text()
    ).toContain('10 minutes')
    expect(
      wrapper.findAll('.measurement-interval-setting select option').length
    ).toBe(4)
    wrapper
      .findAll('.measurement-interval-setting select option')
      .at(0).element.selected = true
    wrapper
      .findAll('.measurement-interval-setting select option')
      .at(1).element.selected = false
    wrapper.find('.measurement-interval-setting select ').trigger('change')

    await wrapper.vm.$nextTick()

    const stringifiedSettings = JSON.stringify(wrapper.vm.settings)
    expect(wrapper.vm.settings.measurement_interval).toBe('5_min')
    expect(patchAccount).toBeCalledWith(store.state.user.access_token, {
      settings: stringifiedSettings
    })
    expect(
      wrapper.find('.measurement-interval-setting .text').text()
    ).toContain('5 minutes')

    expect(wrapper.element).toMatchSnapshot()
  })

  it('wifi settings', async () => {
    const accountResponse = cloneDeep(accountData)
    getAccount.mockResolvedValueOnce(accountResponse)
    const patchResponse = cloneDeep(patchAccountData)
    patchAccount.mockResolvedValue(patchResponse)

    wrapper = factory()

    await wrapper.vm.$nextTick()

    wrapper
      .findAll('.side-panel .list-item')
      .at(2)
      .find('.actions .button')
      .trigger('click')
    expect(wrapper.vm.activeCategoryID).toBe(2)

    const ssidInput = wrapper.find('#wifi_ssid_input')
    ssidInput.trigger('focus')
    ssidInput.setValue('wifi-name')

    const passwordInput = wrapper.find('#wifi_password_input')
    passwordInput.trigger('focus')
    passwordInput.setValue('wifi-password')

    const stringifiedSettings = JSON.stringify(wrapper.vm.settings)
    expect(wrapper.vm.settings.wifi.ssid).toBe('wifi-name')
    expect(wrapper.vm.settings.wifi.password).toBe('wifi-password')
    expect(wrapper.find('.main-panel').text()).toContain(
      'The base station will connect to the \'wifi-name\' wifi network'
    )
    expect(patchAccount).toBeCalledWith(store.state.user.access_token, {
      settings: stringifiedSettings
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('reset token', async () => {
    const accountResponse = cloneDeep(accountData)
    getAccount.mockResolvedValueOnce(accountResponse)

    wrapper = factory()

    await wrapper.vm.$nextTick()

    wrapper
      .findAll('.side-panel .list-item')
      .at(1)
      .find('.actions .button')
      .trigger('click')
    expect(wrapper.vm.activeCategoryID).toBe(1)
    expect(wrapper.find('.settings-box span.bold').text()).toBe(
      store.state.user.reset_token
    )

    expect(wrapper.element).toMatchSnapshot()
  })
})
