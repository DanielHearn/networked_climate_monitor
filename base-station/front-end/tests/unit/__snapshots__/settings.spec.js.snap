// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings.vue category changing 1`] = `
<div
  class="content"
>
  <div
    class="side-panel"
  >
    <div
      class="side-panel-header"
    >
      <p
        class="sub-heading"
      >
        Settings Categories
      </p>
    </div>
     
    <div
      class="side-panel-content"
    >
      <ul
        class="list"
      >
        <li
          class="list-item"
          style="display: flex; flex-direction: column; justify-content: flex-start; text-align: left;"
        >
          <div
            style="display: flex; flex-direction: row;"
          >
            <p
              class="heading"
              style="margin-right: 0.25em; display: flex; align-items: center;"
            >
              
                Measurement Settings
              
            </p>
          </div>
           
          <div
            class="actions"
          >
            <button
              class="button button--primary"
            >
              Edit settings
            </button>
          </div>
        </li>
         
        <li
          class="list-item active"
          style="display: flex; flex-direction: column; justify-content: flex-start; text-align: left;"
        >
          <div
            style="display: flex; flex-direction: row;"
          >
            <p
              class="heading underline"
              style="margin-right: 0.25em; display: flex; align-items: center;"
            >
              
                Account Management
              
            </p>
          </div>
           
          <div
            class="actions"
          >
            <button
              class="button button--primary"
            >
              Manage account
            </button>
          </div>
        </li>
         
        <li
          class="list-item"
          style="display: flex; flex-direction: column; justify-content: flex-start; text-align: left;"
        >
          <div
            style="display: flex; flex-direction: row;"
          >
            <p
              class="heading"
              style="margin-right: 0.25em; display: flex; align-items: center;"
            >
              
                Base Station Management
              
            </p>
          </div>
           
          <div
            class="actions"
          >
            <button
              class="button button--primary"
            >
              Manage base station
            </button>
          </div>
        </li>
      </ul>
    </div>
  </div>
   
  <div
    class="main-panel"
  >
    <div
      class="main-panel-header"
    >
      <p
        class="sub-heading"
      >
        Account Management
      </p>
    </div>
     
    <div
      class="main-panel-content"
    >
      <div
        class="dashboard-content"
      >
        <!---->
         
        <div
          class="settings-box"
        >
          <p
            class="sub-heading"
          >
            Change Password
          </p>
           
          <router-link
            class="link"
            to="/reset-password"
          >
            Change Password
          </router-link>
           
          <p
            class="text italics"
          >
            
                This reset token can be used to reset the account password on
                the password reset page, make sure it is saved somewhere
                secure.
              
          </p>
           
          <p
            class="text"
          >
            
                Reset Token:
                
            <span
              class="bold"
            >
              1234
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Settings.vue change measurement interval 1`] = `
<div
  class="content"
>
  <div
    class="side-panel"
  >
    <div
      class="side-panel-header"
    >
      <p
        class="sub-heading"
      >
        Settings Categories
      </p>
    </div>
     
    <div
      class="side-panel-content"
    >
      <ul
        class="list"
      >
        <li
          class="list-item active"
          style="display: flex; flex-direction: column; justify-content: flex-start; text-align: left;"
        >
          <div
            style="display: flex; flex-direction: row;"
          >
            <p
              class="heading underline"
              style="margin-right: 0.25em; display: flex; align-items: center;"
            >
              
                Measurement Settings
              
            </p>
          </div>
           
          <div
            class="actions"
          >
            <button
              class="button button--primary"
            >
              Edit settings
            </button>
          </div>
        </li>
         
        <li
          class="list-item"
          style="display: flex; flex-direction: column; justify-content: flex-start; text-align: left;"
        >
          <div
            style="display: flex; flex-direction: row;"
          >
            <p
              class="heading"
              style="margin-right: 0.25em; display: flex; align-items: center;"
            >
              
                Account Management
              
            </p>
          </div>
           
          <div
            class="actions"
          >
            <button
              class="button button--primary"
            >
              Manage account
            </button>
          </div>
        </li>
         
        <li
          class="list-item"
          style="display: flex; flex-direction: column; justify-content: flex-start; text-align: left;"
        >
          <div
            style="display: flex; flex-direction: row;"
          >
            <p
              class="heading"
              style="margin-right: 0.25em; display: flex; align-items: center;"
            >
              
                Base Station Management
              
            </p>
          </div>
           
          <div
            class="actions"
          >
            <button
              class="button button--primary"
            >
              Manage base station
            </button>
          </div>
        </li>
      </ul>
    </div>
  </div>
   
  <div
    class="main-panel"
  >
    <div
      class="main-panel-header"
    >
      <p
        class="sub-heading"
      >
        Measurement Settings
      </p>
    </div>
     
    <div
      class="main-panel-content"
    >
      <div
        class="dashboard-content"
      >
        <!---->
         
        <div
          class="input-box measurement-interval-setting"
        >
          <p
            class="sub-heading"
          >
            Node Measurement Interval:
          </p>
           
          <select>
            <option
              value="5_min"
            >
              
                    5 Minutes
                  
            </option>
            <option
              value="10_min"
            >
              
                    10 Minutes
                  
            </option>
            <option
              value="30_min"
            >
              
                    30 Minutes
                  
            </option>
            <option
              value="60_min"
            >
              
                    1 Hour
                  
            </option>
          </select>
           
          <p
            class="text italics"
          >
            
                  Sensor nodes will record climate data every
                  5 minutes
                
          </p>
        </div>
         
        <div
          class="input-box temperature-unit-setting"
        >
          <p
            class="sub-heading"
          >
            Temperature Unit:
          </p>
           
          <div
            class="radio-container"
          >
            <div
              class="radio-option active"
            >
              <label
                for="temp_unit_c"
              >
                Celsius:
                      
                <input
                  class="radio"
                  id="temp_unit_c"
                  type="radio"
                  value="c"
                />
                 
                <span
                  class="input"
                >
                  <i
                    class="material-icons"
                  >
                    done
                  </i>
                </span>
              </label>
            </div>
             
            <div
              class="radio-option"
            >
              <label
                for="temp_unit_f"
              >
                Fahrenheit:
                      
                <input
                  class="radio"
                  id="temp_unit_f"
                  type="radio"
                  value="f"
                />
                 
                <span
                  class="input"
                >
                  <i
                    class="material-icons"
                  />
                </span>
              </label>
            </div>
          </div>
           
          <p
            class="text italics"
          >
            
                  Temperatures are stored in celsius and will be displayed in
                  celsius.
                
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Settings.vue change temperature unit 1`] = `
<div
  class="content"
>
  <div
    class="side-panel"
  >
    <div
      class="side-panel-header"
    >
      <p
        class="sub-heading"
      >
        Settings Categories
      </p>
    </div>
     
    <div
      class="side-panel-content"
    >
      <ul
        class="list"
      >
        <li
          class="list-item active"
          style="display: flex; flex-direction: column; justify-content: flex-start; text-align: left;"
        >
          <div
            style="display: flex; flex-direction: row;"
          >
            <p
              class="heading underline"
              style="margin-right: 0.25em; display: flex; align-items: center;"
            >
              
                Measurement Settings
              
            </p>
          </div>
           
          <div
            class="actions"
          >
            <button
              class="button button--primary"
            >
              Edit settings
            </button>
          </div>
        </li>
         
        <li
          class="list-item"
          style="display: flex; flex-direction: column; justify-content: flex-start; text-align: left;"
        >
          <div
            style="display: flex; flex-direction: row;"
          >
            <p
              class="heading"
              style="margin-right: 0.25em; display: flex; align-items: center;"
            >
              
                Account Management
              
            </p>
          </div>
           
          <div
            class="actions"
          >
            <button
              class="button button--primary"
            >
              Manage account
            </button>
          </div>
        </li>
         
        <li
          class="list-item"
          style="display: flex; flex-direction: column; justify-content: flex-start; text-align: left;"
        >
          <div
            style="display: flex; flex-direction: row;"
          >
            <p
              class="heading"
              style="margin-right: 0.25em; display: flex; align-items: center;"
            >
              
                Base Station Management
              
            </p>
          </div>
           
          <div
            class="actions"
          >
            <button
              class="button button--primary"
            >
              Manage base station
            </button>
          </div>
        </li>
      </ul>
    </div>
  </div>
   
  <div
    class="main-panel"
  >
    <div
      class="main-panel-header"
    >
      <p
        class="sub-heading"
      >
        Measurement Settings
      </p>
    </div>
     
    <div
      class="main-panel-content"
    >
      <div
        class="dashboard-content"
      >
        <!---->
         
        <div
          class="input-box measurement-interval-setting"
        >
          <p
            class="sub-heading"
          >
            Node Measurement Interval:
          </p>
           
          <select>
            <option
              value="5_min"
            >
              
                    5 Minutes
                  
            </option>
            <option
              value="10_min"
            >
              
                    10 Minutes
                  
            </option>
            <option
              value="30_min"
            >
              
                    30 Minutes
                  
            </option>
            <option
              value="60_min"
            >
              
                    1 Hour
                  
            </option>
          </select>
           
          <p
            class="text italics"
          >
            
                  Sensor nodes will record climate data every
                  10 minutes
                
          </p>
        </div>
         
        <div
          class="input-box temperature-unit-setting"
        >
          <p
            class="sub-heading"
          >
            Temperature Unit:
          </p>
           
          <div
            class="radio-container"
          >
            <div
              class="radio-option"
            >
              <label
                for="temp_unit_c"
              >
                Celsius:
                      
                <input
                  class="radio"
                  id="temp_unit_c"
                  type="radio"
                  value="c"
                />
                 
                <span
                  class="input"
                >
                  <i
                    class="material-icons"
                  />
                </span>
              </label>
            </div>
             
            <div
              class="radio-option active"
            >
              <label
                for="temp_unit_f"
              >
                Fahrenheit:
                      
                <input
                  class="radio"
                  id="temp_unit_f"
                  type="radio"
                  value="f"
                />
                 
                <span
                  class="input"
                >
                  <i
                    class="material-icons"
                  >
                    done
                  </i>
                </span>
              </label>
            </div>
          </div>
           
          <p
            class="text italics"
          >
            
                  Temperatures are stored in celsius and will be displayed in
                  farenheit
                
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Settings.vue has the expected html structure 1`] = `
<div
  class="content"
>
  <div
    class="side-panel"
  >
    <div
      class="side-panel-header"
    >
      <p
        class="sub-heading"
      >
        Settings Categories
      </p>
    </div>
     
    <div
      class="side-panel-content"
    >
      <ul
        class="list"
      >
        <li
          class="list-item active"
          style="display: flex; flex-direction: column; justify-content: flex-start; text-align: left;"
        >
          <div
            style="display: flex; flex-direction: row;"
          >
            <p
              class="heading underline"
              style="margin-right: 0.25em; display: flex; align-items: center;"
            >
              
                Measurement Settings
              
            </p>
          </div>
           
          <div
            class="actions"
          >
            <button
              class="button button--primary"
            >
              Edit settings
            </button>
          </div>
        </li>
         
        <li
          class="list-item"
          style="display: flex; flex-direction: column; justify-content: flex-start; text-align: left;"
        >
          <div
            style="display: flex; flex-direction: row;"
          >
            <p
              class="heading"
              style="margin-right: 0.25em; display: flex; align-items: center;"
            >
              
                Account Management
              
            </p>
          </div>
           
          <div
            class="actions"
          >
            <button
              class="button button--primary"
            >
              Manage account
            </button>
          </div>
        </li>
         
        <li
          class="list-item"
          style="display: flex; flex-direction: column; justify-content: flex-start; text-align: left;"
        >
          <div
            style="display: flex; flex-direction: row;"
          >
            <p
              class="heading"
              style="margin-right: 0.25em; display: flex; align-items: center;"
            >
              
                Base Station Management
              
            </p>
          </div>
           
          <div
            class="actions"
          >
            <button
              class="button button--primary"
            >
              Manage base station
            </button>
          </div>
        </li>
      </ul>
    </div>
  </div>
   
  <div
    class="main-panel"
  >
    <div
      class="main-panel-header"
    >
      <p
        class="sub-heading"
      >
        Measurement Settings
      </p>
    </div>
     
    <div
      class="main-panel-content"
    >
      <div
        class="dashboard-content"
      >
        <!---->
         
        <div
          class="input-box measurement-interval-setting"
        >
          <p
            class="sub-heading"
          >
            Node Measurement Interval:
          </p>
           
          <select>
            <option
              value="5_min"
            >
              
                    5 Minutes
                  
            </option>
            <option
              value="10_min"
            >
              
                    10 Minutes
                  
            </option>
            <option
              value="30_min"
            >
              
                    30 Minutes
                  
            </option>
            <option
              value="60_min"
            >
              
                    1 Hour
                  
            </option>
          </select>
           
          <p
            class="text italics"
          >
            
                  Sensor nodes will record climate data every
                  10 minutes
                
          </p>
        </div>
         
        <div
          class="input-box temperature-unit-setting"
        >
          <p
            class="sub-heading"
          >
            Temperature Unit:
          </p>
           
          <div
            class="radio-container"
          >
            <div
              class="radio-option active"
            >
              <label
                for="temp_unit_c"
              >
                Celsius:
                      
                <input
                  class="radio"
                  id="temp_unit_c"
                  type="radio"
                  value="c"
                />
                 
                <span
                  class="input"
                >
                  <i
                    class="material-icons"
                  >
                    done
                  </i>
                </span>
              </label>
            </div>
             
            <div
              class="radio-option"
            >
              <label
                for="temp_unit_f"
              >
                Fahrenheit:
                      
                <input
                  class="radio"
                  id="temp_unit_f"
                  type="radio"
                  value="f"
                />
                 
                <span
                  class="input"
                >
                  <i
                    class="material-icons"
                  />
                </span>
              </label>
            </div>
          </div>
           
          <p
            class="text italics"
          >
            
                  Temperatures are stored in celsius and will be displayed in
                  celsius.
                
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Settings.vue reset token 1`] = `
<div
  class="content"
>
  <div
    class="side-panel"
  >
    <div
      class="side-panel-header"
    >
      <p
        class="sub-heading"
      >
        Settings Categories
      </p>
    </div>
     
    <div
      class="side-panel-content"
    >
      <ul
        class="list"
      >
        <li
          class="list-item"
          style="display: flex; flex-direction: column; justify-content: flex-start; text-align: left;"
        >
          <div
            style="display: flex; flex-direction: row;"
          >
            <p
              class="heading"
              style="margin-right: 0.25em; display: flex; align-items: center;"
            >
              
                Measurement Settings
              
            </p>
          </div>
           
          <div
            class="actions"
          >
            <button
              class="button button--primary"
            >
              Edit settings
            </button>
          </div>
        </li>
         
        <li
          class="list-item active"
          style="display: flex; flex-direction: column; justify-content: flex-start; text-align: left;"
        >
          <div
            style="display: flex; flex-direction: row;"
          >
            <p
              class="heading underline"
              style="margin-right: 0.25em; display: flex; align-items: center;"
            >
              
                Account Management
              
            </p>
          </div>
           
          <div
            class="actions"
          >
            <button
              class="button button--primary"
            >
              Manage account
            </button>
          </div>
        </li>
         
        <li
          class="list-item"
          style="display: flex; flex-direction: column; justify-content: flex-start; text-align: left;"
        >
          <div
            style="display: flex; flex-direction: row;"
          >
            <p
              class="heading"
              style="margin-right: 0.25em; display: flex; align-items: center;"
            >
              
                Base Station Management
              
            </p>
          </div>
           
          <div
            class="actions"
          >
            <button
              class="button button--primary"
            >
              Manage base station
            </button>
          </div>
        </li>
      </ul>
    </div>
  </div>
   
  <div
    class="main-panel"
  >
    <div
      class="main-panel-header"
    >
      <p
        class="sub-heading"
      >
        Account Management
      </p>
    </div>
     
    <div
      class="main-panel-content"
    >
      <div
        class="dashboard-content"
      >
        <!---->
         
        <div
          class="settings-box"
        >
          <p
            class="sub-heading"
          >
            Change Password
          </p>
           
          <router-link
            class="link"
            to="/reset-password"
          >
            Change Password
          </router-link>
           
          <p
            class="text italics"
          >
            
                This reset token can be used to reset the account password on
                the password reset page, make sure it is saved somewhere
                secure.
              
          </p>
           
          <p
            class="text"
          >
            
                Reset Token:
                
            <span
              class="bold"
            >
              1234
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Settings.vue wifi settings 1`] = `
<div
  class="content"
>
  <div
    class="side-panel"
  >
    <div
      class="side-panel-header"
    >
      <p
        class="sub-heading"
      >
        Settings Categories
      </p>
    </div>
     
    <div
      class="side-panel-content"
    >
      <ul
        class="list"
      >
        <li
          class="list-item"
          style="display: flex; flex-direction: column; justify-content: flex-start; text-align: left;"
        >
          <div
            style="display: flex; flex-direction: row;"
          >
            <p
              class="heading"
              style="margin-right: 0.25em; display: flex; align-items: center;"
            >
              
                Measurement Settings
              
            </p>
          </div>
           
          <div
            class="actions"
          >
            <button
              class="button button--primary"
            >
              Edit settings
            </button>
          </div>
        </li>
         
        <li
          class="list-item"
          style="display: flex; flex-direction: column; justify-content: flex-start; text-align: left;"
        >
          <div
            style="display: flex; flex-direction: row;"
          >
            <p
              class="heading"
              style="margin-right: 0.25em; display: flex; align-items: center;"
            >
              
                Account Management
              
            </p>
          </div>
           
          <div
            class="actions"
          >
            <button
              class="button button--primary"
            >
              Manage account
            </button>
          </div>
        </li>
         
        <li
          class="list-item active"
          style="display: flex; flex-direction: column; justify-content: flex-start; text-align: left;"
        >
          <div
            style="display: flex; flex-direction: row;"
          >
            <p
              class="heading underline"
              style="margin-right: 0.25em; display: flex; align-items: center;"
            >
              
                Base Station Management
              
            </p>
          </div>
           
          <div
            class="actions"
          >
            <button
              class="button button--primary"
            >
              Manage base station
            </button>
          </div>
        </li>
      </ul>
    </div>
  </div>
   
  <div
    class="main-panel"
  >
    <div
      class="main-panel-header"
    >
      <p
        class="sub-heading"
      >
        Base Station Management
      </p>
    </div>
     
    <div
      class="main-panel-content"
    >
      <div
        class="dashboard-content"
      >
        <!---->
         
        <div
          class="input-box measurement-interval-setting"
        >
          <p
            class="sub-heading"
          >
            Wifi Settings
          </p>
           
          <input
            class="input--text input--small"
            id="wifi_ssid_input"
            placeholder="Wifi SSID"
            type="text"
          />
           
          <input
            class="input--text input--small"
            id="wifi_password_input"
            placeholder="Wifi Password"
            type="text"
          />
           
          <p>
            
                The base station will connect to the 'wifi-name' wifi network if it is available, otherwise it will try to
                connect to the 'climate-monitor' wifi network.
              
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
